spec/controllers/spree/api/line_items_controller_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'spec_helper'
+
+module Spree
+  describe Spree::Api::LineItemsController do
+    render_views
+
+    before do
+      stub_authentication!
+      Spree.user_class.stub :find_by_spree_api_key => current_api_user
+    end
+
+    def self.make_simple_data!
+      let!(:order) { FactoryGirl.create(:order, state: 'complete', completed_at: Time.now) }
+      let!(:line_item) { FactoryGirl.create(:line_item, order: order, unit_value: 500) }
+    end
+
+    #test that when a line item is updated, an order's fees are updated too
+    context "as an admin user" do
+      sign_in_as_admin!
+      make_simple_data!
+
+      context "as a line item is updated" do
+        it "apply enterprise fees on the item" do
+          line_item_params = { order_id: order.number, id: line_item.id, line_item: { id: line_item.id, unit_value: 520 }, format: :json}
+          controller.should_receive(:apply_enterprise_fees).and_return(true)
```

Comment:
Hi @lin-d-hop,

I really think that you can just ditch the apply_enterprise_fees stub on the controller, and just have one for update_distribution_charge! on the order.

If that doesn't work, maybe push what you have into this branch and I'll have a play around and see if I can understand what is happening.